great_expectations_cloud/agent/actions/generate_schema_change_expectations_action.py

```diff
@@ -52,30 +58,87 @@ def __init__(  # noqa: PLR0913  # Refactor opportunity
 
     @override
     def run(self, event: GenerateSchemaChangeExpectationsEvent, id: str) -> ActionResult:
-        metric_run_id = uuid.uuid4()
-
-        for asset in event.data_assets:
-            datasource = self._context.data_sources.get(event.datasource_name)
-            data_asset = datasource.get_asset(asset)
-            data_asset.test_connection()
-            batch_request = data_asset.build_batch_request()
-
-            self._batch_inspector.compute_metric_list_run(
-                data_asset_id=data_asset.id,
-                batch_request=batch_request,
-                metric_list=[MetricTypes.TABLE_COLUMN_TYPES, MetricTypes.TABLE_COLUMNS],
-            )
-            # TODO - Update with full logic in ZELDA-1058
-
-        # TODO - Update this after the full implementation in ZELDA-1058
+        created_resources: list[CreatedResource] = []
+        for asset_name in event.data_assets:
+            try:
+                data_asset = self._retrieve_asset_from_asset_name(event, asset_name)
+                metric_run, metric_run_id = self._get_metrics(data_asset)
+                # this logic will change with ZELDA-1154 and we will be pulling this in from a mapping
+                expectation_suite_name = f"{data_asset.name} - Expectation Suite"
+                expectation = self._add_schema_change_expectation(
+                    metric_run, expectation_suite_name
+                )
+                created_resources.append(
+                    CreatedResource(resource_id=str(metric_run_id), type="MetricRun")
+                )
+                created_resources.append(
+                    CreatedResource(resource_id=expectation.id, type="Expectation")
+                )
+            except Exception as e:
+                # TODO - follow up in ZELDA-1153
+                LOGGER.warning(f"asset_name: {asset_name} failed with error: {e}")
+                pass
         return ActionResult(
             id=id,
             type=event.type,
-            created_resources=[
-                CreatedResource(resource_id=str(metric_run_id), type="MetricRun"),
-            ],
+            created_resources=created_resources,
         )
 
+    def _retrieve_asset_from_asset_name(
+        self, event: GenerateSchemaChangeExpectationsEvent, asset_name: str
+    ) -> DataAsset:
+        try:
+            datasource = self._context.data_sources.get(event.datasource_name)
+            data_asset = datasource.get_asset(asset_name)
+            data_asset.test_connection()  # raises `TestConnectionError` on failure
+
+        except Exception as e:
+            # TODO - see if this can be made more specific
+            raise RuntimeError(f"Failed to retrieve asset: {e}") from e  # noqa: TRY003 # want to keep this informative for now
```

Comment:
What do you think about removing this try/except here? The try/except in the run method will catch any error raised here, and will report the asset name and error (line 79).
Eventually in ZELDA-1153 we can collect any raised exceptions and re-raise in the except block of the `run` method so that the event/action handling thread can catch it and report it back to the backend.

Reply:
Actually, I see how you are raising other errors in the methods that get executed in the `run` method to be descriptive about which part of the process failed. That makes sense. Maybe we can consider more custom error types rather than `RuntimeError` e.g. `AssetRetrievalError` but I'm ok with this for now.